### ISSUE TITLE
2020-04-03 tc111 M4 RATEN returning NaN instead of NA when mct$ACTENDTIME <= mct$ACTTIME

### ISSUE THREAD

**Issue by jhhughes256** [tensfeldt/openNCA#205]
### Issue

When using M4 with the same data as tc111 using actual time, RATEN generates NaN instead of NA for time intervals of 0 to 0. When mct$ACTENDTIME <= mct$ACTTIME these values should be NA as per the CE specificaiton.

### Example

Example not currently possible for tc111 is not currently running in my environment.

Instead tc1574 is used for example below.

```r
> refparams[,rateincrements]
         RATE1       RATE2      RATE3     RATE4     RATE5    RATE6    RATE7    RATE8   RATE9 RATE10 RATE11 RATE12
1           NA 70080798873 2617858333 457290226 188227313 97299583 51492756 20066822 5149558      0     NA     NA
2  83257635207  3061084649  555427000 223162094 106764625 43295101 24461247       NA      NA     NA     NA     NA
3           NA 74043192177 2717823333 478299134  72202400  3664495        0        0       0      0     NA     NA
4  80068008850  3771161646  550186333 107930528   7060669  2353973        0       NA      NA     NA     NA     NA
5           NA 79388348388 1800456059 229144721  31694293  9331213  5951639        0       0      0     NA     NA
6  84820493274  2596560000  299560750   8103697         0        0        0       NA      NA     NA     NA     NA
7           NA 72304831358 1796677763 376472457 159209102 64103131 24470981  4006934       0      0     NA     NA
8  83416377171  2272158892  356469000  81603146  32072025 19809638 13534265       NA      NA     NA     NA     NA
9           NA 83204937759 1852970791 115892708  21481725  3660579  2541203        0       0      0     NA     NA
10 84573289614  2529358258  333412916 116733553  53678035 23850267  5273470       NA      NA     NA     NA     NA
11          NA 68199767307 1937724890 324800000  33365688        0        0        0       0      0      0      0
> results[,rateincrements]
         RATE1       RATE2      RATE3     RATE4     RATE5    RATE6    RATE7    RATE8   RATE9 RATE10 RATE11 RATE12
1          NaN 70080798873 2617858333 457290226 188227313 97299583 51492756 20066822 5149558      0     NA     NA
2  83257635207  3061084649  555427000 223162094 106764625 43295101 24461247       NA      NA     NA     NA     NA
3          NaN 74043192177 2717823333 478299134  72202400  3664495        0        0       0      0     NA     NA
4  80068008850  3771161646  550186333 107930528   7060669  2353973        0       NA      NA     NA     NA     NA
5          NaN 79388348388 1800456059 229144721  31694293  9331213  5951639        0       0      0     NA     NA
6  84820493274  2596560000  299560750   8103697         0        0        0       NA      NA     NA     NA     NA
7          NaN 72304831358 1796677763 376472457 159209102 64103131 24470981  4006934       0      0     NA     NA
8  83416377171  2272158892  356469000  81603146  32072025 19809638 13534265       NA      NA     NA     NA     NA
9          NaN 83204937759 1852970791 115892708  21481725  3660579  2541203        0       0      0     NA     NA
10 84573289614  2529358258  333412916 116733553  53678035 23850267  5273470       NA      NA     NA     NA     NA
11         NaN 68199767307 1937724890 324800000  33365688        0        0        0       0      0      0      0
```

**Comment by jhhughes256** [comment]
This issue still occurs as of commit ed0aed3 and can now be tested with tc111.

**Comment by jhhughes256** [comment]
This issue has been resolved for tc111 and tc1574 with commit cca452d. This issue will be closed.

### tc111 ed0aed3

```r
knitr::kable(r_old[,paste0("RATE", 1:12)])
```

| RATE1|    RATE2|    RATE3|    RATE4|     RATE5|     RATE6|     RATE7|     RATE8|    RATE9| RATE10| RATE11| RATE12|
|-----:|--------:|--------:|--------:|---------:|---------:|---------:|---------:|--------:|------:|------:|------:|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|   NaN| 70080.80| 2617.858| 457.2902| 188.22731| 97.299583| 51.492756| 20.066822| 5.149558|      0|     NA|     NA|
|   NaN| 83204.94| 1852.971| 115.8927|  21.48173|  3.660579|  2.541202|  0.000000| 0.000000|      0|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|   NaN| 74043.19| 2717.823| 478.2991|  72.20240|  3.664496|  0.000000|  0.000000| 0.000000|      0|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|   NaN| 79388.35| 1800.456| 229.1447|  31.69429|  9.331213|  5.951639|  0.000000| 0.000000|      0|     NA|     NA|
|   NaN| 68199.77| 1937.725| 324.8000|  33.36569|  0.000000|  0.000000|  0.000000| 0.000000|      0|      0|      0|
|   NaN| 72304.83| 1796.678| 376.4725| 159.20910| 64.103131| 24.470981|  4.006933| 0.000000|      0|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|

### tc111 cca452d

```r
knitr::kable(r[,paste0("RATE", 1:12)])
```

| RATE1|    RATE2|    RATE3|    RATE4|     RATE5|     RATE6|     RATE7|     RATE8|    RATE9| RATE10| RATE11| RATE12|
|-----:|--------:|--------:|--------:|---------:|---------:|---------:|---------:|--------:|------:|------:|------:|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|    NA| 70080.80| 2617.858| 457.2902| 188.22731| 97.299583| 51.492756| 20.066822| 5.149558|      0|     NA|     NA|
|    NA| 83204.94| 1852.971| 115.8927|  21.48173|  3.660579|  2.541202|  0.000000| 0.000000|      0|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|    NA| 74043.19| 2717.823| 478.2991|  72.20240|  3.664496|  0.000000|  0.000000| 0.000000|      0|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
|    NA| 79388.35| 1800.456| 229.1447|  31.69429|  9.331213|  5.951639|  0.000000| 0.000000|      0|     NA|     NA|
|    NA| 68199.77| 1937.725| 324.8000|  33.36569|  0.000000|  0.000000|  0.000000| 0.000000|      0|      0|      0|
|    NA| 72304.83| 1796.678| 376.4725| 159.20910| 64.103131| 24.470981|  4.006933| 0.000000|      0|     NA|     NA|
|    NA|       NA|       NA|       NA|        NA|        NA|        NA|        NA|       NA|     NA|     NA|     NA|
